bundle.go

```
@@ -0,0 +1,151 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	spec "github.com/opencontainers/specs/specs-go"
+)
+
+const image = "busybox"
+
+const tmpDir = "/tmp"
+
+const configTemplate = "src/github.com/clearcontainers/tests/data/config.json"
+
+// Bundle represents the root directory where config.json and rootfs are
+type Bundle struct {
+	// Config represents the config.json
+	Config *spec.Spec
+
+	// Path to the bundle
+	Path string
+}
+
+// NewBundle creates a new bundle
+func NewBundle(workload []string) (*Bundle, error) {
+	path, err := ioutil.TempDir(tmpDir, "bundle")
+	if err != nil {
+		return nil, err
+	}
+
+	if err := createRootfs(path); err != nil {
+		return nil, err
+	}
+
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		return nil, fmt.Errorf("GOPATH is not set")
+	}
+
+	configTemplatePath := filepath.Join(gopath, configTemplate)
+	content, err := ioutil.ReadFile(configTemplatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config.json template: %s", err)
+	}
+
+	var config spec.Spec
+	err = json.Unmarshal(content, &config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config.json template: %s", err)
+	}
+
+	config.Process.Args = workload
+
+	bundle := &Bundle{
+		Path:   path,
+		Config: &config,
+	}
+
+	err = bundle.Save()
+	if err != nil {
+		return nil, err
+	}
+
+	return bundle, nil
+}
+
+// createRootfs creates a rootfs in the specific bundlePath
+func createRootfs(bundlePath string) error {
```

Maybe return an error immediately if `bundlePath == ""`?